iwyu_ast_util.cc

```diff
@@ -165,6 +165,20 @@ TemplateSpecializationKind GetTemplateSpecializationKind(const Decl* decl) {
   return clang::TSK_Undeclared;
 }
 
+const Type* DesugarAliasTypes(const Type* type) {
```

Comment:
This function can be `static`.

It would be helpful with a comment to describe what the subtle difference is between this and a full clang desugar vs. a normal IWYU Desugar. This basically desugars everything except non-alias template specializations, right?

Reply:
> This function can be static.

What for? It is inside an anonymous namespace, so already has internal linkage.

Reply:
> This basically desugars everything except non-alias template specializations, right?

Yes, it is. I can leave a comment.

Reply:
Ah, I see. The problem with anonymous namespaces is exactly this: you can't see on the declaration whether it's external/internal, you need to find the (usually very distant) enclosing scope. See also https://llvm.org/docs/CodingStandards.html#anonymous-namespaces.

But not a lot of functions in this file are explicitly marked static, so as long as it's internal linkage, I'm happy with it as is.

Reply:
Up to you, there's no strong consistency anyway :)